src/test/regress/sql/multi_test_helpers_superuser.sql

```diff
@@ -30,9 +30,17 @@ WITH dist_node_summary AS (
         master_run_on_worker(ARRAY[hostname, 'localhost'], ARRAY[port, master_port],
                             ARRAY[dist_placement_summary.query, dist_placement_summary.query],
                             false)
+), dist_object_summary AS (
+    SELECT 'SELECT jsonb_agg(ROW(classid, objid, objsubid, distribution_argument_index, colocationid) ORDER BY classid, objid, objsubid) FROM pg_dist_object' AS query
+), dist_object_check AS (
+    SELECT count(distinct result) = 1 AS matches
+    FROM dist_object_summary CROSS JOIN LATERAL
+        master_run_on_worker(ARRAY[hostname, 'localhost'], ARRAY[port, master_port],
+                            ARRAY[dist_object_summary.query, dist_object_summary.query],
+                            false)
 )
-SELECT dist_node_check.matches AND dist_placement_check.matches
-FROM dist_node_check CROSS JOIN dist_placement_check
+SELECT dist_node_check.matches AND dist_placement_check.matches AND dist_object_check.matches
```

Comment:
this query is non-trivial. Did you manually verify that if the objects are not in sync (e.g., manually pg_dist_object) and show that this returns false?

Also, how come we don't use `citus.pg_dist_object` but only `pg_dist_object`?